src/5.0_to_5.1/migrateContextValues.ts

```diff
@@ -0,0 +1,11 @@
+import { QueryContextEntry } from "@activeviam/activeui-sdk-5.0";
+
+/**
+ * Stringifes the values of the entries in `queryContext`.
+ * Mutates `queryContext`.
+ */
+export function migrateContextValues(queryContext?: QueryContextEntry[]): void {
```

Comment:
```suggestion
export function migrateContextValues(queryContext: DashboardState50["queryContext"] = []): void {
```
There are two suggestions here:
- using `DashboardState50["queryContext"]` instead of `QueryContextEntry[]`: I find that weird for a function to accept a single parameter as being optional. As I read it in its current state, the role of this function is to migrate the values of a query context, so if there is no queryContext, there's no point in calling the function at all. However this function will always be applied to the `queryContext` part of the dashboard state, which will always be typed as potentially undefined, so I agree it's better reflected and handled directly here. And I find this better reflected typed with `DashboardState50["queryContext"]`.
- defaulting the value as an empty array. That way `queryContext` is always an array and you wouldn't use `?.` after `queryContext`, which I find more correct. But honestly it's more of a preference than something I feel strongly about, so feel free not to apply this suggestion.

Reply:
I find the second option misleading because it applies not just to the state of a dashboard, but also of a widget and a page.
And doing `DashboardState50["queryContext"] | DashboardPageState50["queryContext"] | AWidgetState50["queryContext"]` looks like turning something simple into something complicated.
I think this is a good reflection of why, in general, variables should be named and typed after their nature rather than after how they're used.

The second option would mean adding an unnecessary empty array in the saved content.
It's not a huge deal but if we do that for many pieces of state and for 10 000 saved items it's a shame.

  With that in mind, having an optional argument is acceptable imo.

Reply:
Yeah, you're raising good points I hadn't thought about. I'm happy to keep it this way, it's a detail anyway.

I'd just like your opinion: in general, don't you think it's weird to have such a parameter of a function to be optional? Like here the only role of the function is to mutate its parameter. Like there is in theory no point in calling the function at all if this parameter is not here. To me, it's not the role of the function to handle the case where its parameter would be undefined, it shouldn't care about it. Is it just me who is nitpicky about this or do you agree (again, I'm talking in a general way, not in this specific case)?

Reply:
It depends a bit on the usage imo:  the pro is to factorize the handling of the edge case, and in simple cases like this I think pragmatically it's a good enough argument.
But when the call stack grows big, then I think it's wrong to delegate the handling of the edge case several times down the call stack => edge cases should be handled as soon as possible.

So in general I agree yes, and in fact I find it very frustrating that you don't have another choice than the optional arg pattern with React hooks for example.